src/Storages/StorageMergeTree.cpp

```diff
@@ -984,6 +983,13 @@ std::shared_ptr<StorageMergeTree::MergeMutateSelectedEntry> StorageMergeTree::se
         tagger = std::make_unique<CurrentlyMergingPartsTagger>(future_part, MergeTreeDataMergerMutator::estimateNeededDiskSpace({part}), *this, metadata_snapshot, true);
         return std::make_shared<MergeMutateSelectedEntry>(future_part, std::move(tagger), commands);
     }
+
+    if (are_some_mutations_for_some_parts_skipped)
```

Comment:
Maybe this triggering can this be done unconditionally?
Since if some parts was affected then it will not hit this section anyway, right?

Reply:
I intentionally made that in order to not degrade performance on mutex each merge attempt for those who don't use this feature (who mutate always all data). Do you think it worth making unconditionally?